packages/cozy-stack-client/src/AppCollection.js

```diff
@@ -35,8 +36,16 @@ class AppCollection extends DocumentCollection {
     }
   }
 
-  async get() {
-    throw new Error('get() method is not yet implemented')
+  /**
+   * Get an app by id
+   *
+   * @param  {string} id The document id.
+   * @returns {object}  JsonAPI response containing normalized document as data attribute
+   */
+  async get(id) {
+    return Collection.get(this.stackClient, `${this.endpoint}${id}`, {
+      normalize: this.constructor.normalizeDoctype(this.doctype)
+    })
```

Comment:
Instead of copy/pasting the method get here (from DocumentCollection), I think it would be better (in the sense that it would be more DRY) to have :

```
    const endpoint = this.endpoint || `/data/${this.doctype}/`
    return Collection.get(this.stackClient, uri`${this.endpoint}${id}`, {
      normalize: this.constructor.normalizeDoctype(this.doctype)
    })
```

in DocumentCollection.get.

What do you think ?

I see that the `uri` function is not used on the template string also, is there a particular reason for that ?

Reply:
Better, maybe DocumntCollection could set `this.endpoint` in its constructor. This would remove the repetition of `/data/${this.doctype}` in the DocumentCollection code and would make it symmetrical with AppCollection.

Reply:
Agreed with that. The only thing is, I didn't use `uri` because the `/apps/id` was encoded in `%2Fapps%2Fid`, which was failing with a `'400 Bad Request'`.
Any idea why?

Reply:
Yes. uri works on every part of the template literal expressions. So in

```
uri`/data/${urlPart}`
```

Only ${urlPart} will go through the encodeURIComponent. When ${endpoint} goes inside, it gets uri-encoded.

See https://github.com/cozy/cozy-client/blob/7e38f5b06493184c102c7ace2ffab5965e5249b8/packages/cozy-stack-client/src/utils.js#L1 for the implementation.

So `uri` should not be applied to all the URL if ${endpoint} is used. I am not sure `uri` is very ergonomic/explicit here. It is easy to misuse. Feel free to use another technique to be sure that parts that need to be encoded are encoded ;)

Reply:
Ok, thanks for the literal insights! See the [new commit](058e239a811f60839ce95cc764a8b8fa843b7640)